File: Benchmarks_FSharp/Core/Systems/Contract/IRenderSystem.fs
namespace Benchmarks.Core.Systems.Contract

open Benchmarks.Core.Types

[<AbstractClass>]
type IRenderSystem() =
    inherit ISystem()

    abstract Render2D: GameTime -> unit
    default _.Render2D(_) = ()

    abstract Render3D: GameTime -> unit
    default _.Render3D(_) = ()

File: Benchmarks/Core/Systems/Contract/IRenderSystem.cs
using Benchmarks.Core.Types;

namespace Benchmarks.Core.Systems.Contract
{
    /// <summary>
    /// Sistema base para sistemas que renderizan.
    /// Sin dependencia a SpriteBatch - simula operaciones de render.
    /// </summary>
    public abstract class IRenderSystem : ISystem
    {
        public virtual void Render2D(GameTime gameTime) { }
        public virtual void Render3D(GameTime gameTime) { }
    }
}
